src/IbexaPlatformAssetsBundle/lib/Resolver/IconPathResolver.php

```diff
@@ -9,32 +9,59 @@
 namespace Ibexa\Platform\Assets\Resolver;
 
 use eZ\Publish\Core\MVC\ConfigResolverInterface;
+use eZ\Publish\Core\MVC\Symfony\Event\ScopeChangeEvent;
+use eZ\Publish\Core\MVC\Symfony\MVCEvents;
 use Symfony\Component\Asset\Packages;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
 
 /**
  * @internal
  */
-final class IconPathResolver implements IconPathResolverInterface
+final class IconPathResolver implements IconPathResolverInterface, EventSubscriberInterface
 {
     /** @var \eZ\Publish\Core\MVC\ConfigResolverInterface */
     private $configResolver;
 
     /** @var \Symfony\Component\Asset\Packages */
     private $packages;
 
+    /** @var string[] */
+    private $iconCache;
+
     public function __construct(
         ConfigResolverInterface $configResolver,
         Packages $packages
     ) {
         $this->configResolver = $configResolver;
         $this->packages = $packages;
+        $this->iconCache = [];
     }
 
     public function resolve(string $icon, ?string $set = null): string
     {
+        if (array_key_exists($set, $this->iconCache)) {
+            if (array_key_exists($icon, $this->iconCache[$set])) {
+                return $this->iconCache[$set][$icon];
+            }
+        }
```

Comment:
This is a simpler variant which does not involve checking across multiple layers of the array.
```suggestion
        if (isset($this->iconCache[$set][$icon])) {
            return $this->iconCache[$set][$icon];
        }

```

Note that this works because we never expect a `null` value in the array.

Also a whitespace is added to show that it is a completely different part of the operation.

Alternative approach - which you can find in most Symfony code - is to instead have an initializing code wrapped with `isset(...)` condition. This way it ensures that always the value from "cache" property is returned. But the end result is the same as your code and both are completely understandable, so it's :+1: .